test/include/seqan3/test/sandboxed_path.hpp

```diff
@@ -0,0 +1,334 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/* \file
+ * \brief Provides the sandboxed_path and surrounding free functions
+ * \author Simon Gene <simon.gottlieb AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/filesystem>
+
+namespace seqan3::test
+{
+
+/** \brief sandboxed_path is a path that can not leave a certain path
+ *
+ * sandboxed_path inherits from std::filesystem::path and behaves mostly
+ * like it. In addition it receives a sandbox directory at construction time.
+ * Functions are overloaded and a check for the invariant is added.
+ *
+ *  Invariant:
+ *  - sandboxed_path is always converted to an absolute path
+ *  - sandboxed_path always points to a file inside a given sandbox directory.
+ *  - sandbox directory is unmutable during the life cycle of a sandboxed_path
+ *
+ * Caveat:
+ *  - relatives paths are not possible
+ *  - sandboxed_path is not 100% tight
+ *    - calling relative_path() leaves the environment of sandboxed_path
+ */
+class sandboxed_path : public std::filesystem::path
+{
+private:
+    std::filesystem::path const sandbox_path;
+
+public:
+    /*!\brief Construction of a sandboxed_path
+     * \param path must be an absolute path.
+     *
+     * A sandboxed_path initialized with this constructor will
+     * point to path and disallow extension that leave path.
```

Comment:
maybe "disallow an extension that leaves the path" or "disallow extensions that leave the  path" is a bit clearer.